source/diffHandler.py

```diff
@@ -93,15 +94,16 @@ def diff(self, newText: str, oldText: str) -> List[str]:
 					for line in buf.decode("utf-8").splitlines()
 					if line and not line.isspace()
 				]
-			except Exception:
-				log.exception("Exception in DMP, falling back to difflib")
-				return Difflib().diff(newText, oldText)
+		except Exception:
+			log.exception("Exception in DMP, falling back to difflib")
+			return Difflib().diff(newText, oldText)
 
 	def _terminate(self):
 		if DiffMatchPatch._proc:
 			log.debug("Terminating diff-match-patch proxy")
 			# nvda_dmp exits when it receives two zero-length texts.
-			DiffMatchPatch._proc.stdin.write(struct.pack("=II", 0, 0))
+			with DiffMatchPatch._lock:
+				DiffMatchPatch._proc.stdin.write(struct.pack("=II", 0, 0))

```

Comment:
- access to `_proc` should be inside the lock, this goes for initialize also, otherwise the check for proc can be truthy meanwhile terminate is called. 
- This doesn't actually confirm the process exits, perhaps it should?
- `_proc` should be reset to `None`